common/components/ui/Modal.jsx

```diff
@@ -0,0 +1,107 @@
+// @flow
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import closeIcon from 'assets/images/icon-x.svg';
+
+import './Modal.scss';
+
+export default class Modal extends Component {
+  static propTypes = {
+    isOpen: PropTypes.bool,
```

Comment:
Dunno, this makes implementation/usage simpler. Moving back to add animations could be tough tho :D